### PR TITLE
Fix implementation to avoid Object.prototype mutation, + perf improvement

### PR DESCRIPTION
**Pull request by timruffles**
- use a simple function instead
- also bumped the minor version as the public API expanded

### CODE REVIEW

**Comment by timruffles** [comment]
@omnidan also removed an unnecessary nested loop, as the `getKeyByValue` thing is more efficiently implemented by inverting the Emoji map.

**Comment by smeijer** [comment]
Awesome; I just started a PR, but I'll cancel that one 😁

File: lib/emoji.js [hunk 1]
```diff
@@ -126,3 +127,25 @@ Emoji.search = function search(str) {
...
+}, {});
+
+/**
+ * unemojify a string (replace emoji with :emoji:)
+ * @param  {string} str
+ * @return {string}
+ */
+Emoji.unemojify = function unemojify(str) {
+  if (!str) return '';
+  var words = toArray(str);
```
Comment (shvaikalesh) [inline lib/emoji.js]:
```js
var words = Array.from(str);
```

//cc @paulmillr

Comment (bazyli-brzoska) [inline lib/emoji.js]: Indeed, this works properly with Emojis:

```js
> const emoji = '😍🌸'
undefined
> toArr = require('lodash').toArray
[Function: toArray]
> toArr(emoji) // lodash version
[ '😍', '🌸' ]
> [].slice.call(emoji) // slice version
[ '�', '�', '�', '�' ]
> Array.from(emoji) // yeey! no need for the dependency
[ '😍', '🌸' ]
```

Comment (shivkanthb) [inline lib/emoji.js]: Does not work for complex emojis. Try for something like 👩‍👩‍👧‍👧. Thats why I used the lodash library.

Comment (bazyli-brzoska) [inline lib/emoji.js]: Aah 😢. Lodash maintainers did a good job with their implementation then!

**Comment by omnidan** [comment]
thank you so much for all the effort @timruffles ! this PR looks good and I'll merge it!

**Comment by omnidan** [comment]
published `1.7.0` with this merged

**Pull request merged**